Automattic/jetpack
# Fix blogroll block unhelpful error when user not connected

## Proposed changes:

* Create a new shared block component that is specific for connection nudges
* Replace the specific connection banner for Jetpack AI with the shared block
* Add new shared block to blogroll to show connection nudge when user is not connected instead of error

### Other information:


## Jetpack product discussion

P2: pbNhbs-bHl-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Checkout this branch via the Jetpack Beta plugin or your local dev environment
2. With a site that is site-connected but not user connected, create a new blog post
3. Try to add the Blogroll block to the post
4. You should be met with a connection nudge instead of an unhelpful red error
![image](https://github.com/user-attachments/assets/99170b2a-1739-4838-ab81-f58b62423804)
5. Try to add the AI block as well and ensure the connection nudges look the same (there are slight differences as I didn't want to affect the AI nudge)
![image](https://github.com/user-attachments/assets/86bfb9a0-3278-4cad-a3d3-1a7e3e14265f)
6. Click the CTA on the blogroll connection nudge and connect your user account. Go back to the blog post and ensure the blogroll block works as expected
![image](https://github.com/user-attachments/assets/99d8c4d6-796e-4528-b3a4-d3261f9f72c1)

Bonus!

If you'd like, you can try connecting a one user to your site and edit the blog post with a different one. Any connected user account should allow the block to be used.
